libs/labelbox/src/labelbox/data/serialization/ndjson/label.py

```diff
@@ -24,23 +23,68 @@
 from .objects import NDObject, NDObjectType, NDSegments, NDDicomSegments, NDVideoMasks, NDDicomMasks
 from .relationship import NDRelationship
 from .base import DataRow
+from pydantic import BaseModel, ConfigDict, model_serializer, ValidationError
+from .base import subclass_registry, SubclassRegistryBase, NDAnnotation
+from pydantic_core import PydanticUndefined
+from pydantic.alias_generators import to_camel
+from contextlib import suppress
 
 AnnotationType = Union[NDObjectType, NDClassificationType, NDPromptClassificationType,
                        NDConfusionMatrixMetric, NDScalarMetric, NDDicomSegments,
                        NDSegments, NDDicomMasks, NDVideoMasks, NDRelationship,
                        NDPromptText]
 
 
-class NDLabel(pydantic_compat.BaseModel):
-    annotations: List[AnnotationType]
+class NDLabel(BaseModel):
+    annotations: List[SubclassRegistryBase]
+    
+    def __init__(self, **kwargs):
```

Comment:
This is the workaround I did for a hack we had. View the article in the comments for more details. Prior the order of the subclasses made them serialize correct this greatly simplifies the library TBH

Reply:
what was the hack?

Reply:
DMd you but it was this [hack](https://github.com/Labelbox/labelbox-python/blob/906c2e912aa13cfa6242fa417df5df7bdd866e7f/libs/labelbox/src/labelbox/data/serialization/ndjson/objects.py#L776-L778). This is by far the most complicated solution I had to create. I tried to include a lot of comments